README.md

```diff
@@ -502,6 +503,43 @@ class MyGem::Inflector < Zeitwerk::GemInflector
 end
 ```
 
+<a id="markdown-the-on_load-callback" name="the-on_load-callback"></a>
+### The `on_load` callback
+
+The usual place to run something when a file is loaded is the file itself. However, sometimes you'd like to be called, and this is possible with the `on_load` callback.
+
+For example, let's say a class managed by Zeitwerk in a web application has a class-level attribute whose value depends on the environment:
+
+```ruby
+class SomeApiClient
+  class << self
+    attr_accessor :endpoint
+  end
+end
+```
+
+With `on_load`, it is easy to initialize said attribute at boot time according to the configuration:
```

Comment:
Maybe we could mention somewhere that it's especially useful because it avoids loading a constant if you didn't need to?

Reply:
Good, I'll add something in that line. It also a way to avoid autoloading during boot time in Rails, and in a way that works well with reloading.

I'll give it a thought.

Reply:
I've written this:

> Uses cases:
> 
> * Doing something with an autoloadable class or module in a Rails application during initialization, in a way that plays well with reloading.
> * Delaying the execution of the block until the class is loaded for performance.
> * Delaying the execution of the block until the class is loaded because it follows the adapter pattern and better not to load the class if the user does not need it.
> * Etc.
> 
> However, let me stress that the easiest way to accomplish that is to put the code in the actual file. `on_load` is useful when from a design point of view it is better that the code in the callback is elsewhere.
> 

Going to wrap up and publish a new version soon.